contracts/test/vault/oeth-swapper.js

```diff
@@ -0,0 +1,415 @@
+      const fromAmount = utils.parseEther("100");
+      const toAmount = utils.parseEther("100");
+
+      // Call swap method
+      const tx = oethVault
+        .connect(strategist)
+        .swapCollateral(dai.address, weth.address, fromAmount, toAmount, []);
+
+      await expect(tx).to.be.revertedWith("From asset is not supported");
+    });
+
+    it("Should revert if toAsset is not supported", async () => {
+      const { weth, dai, oethVault, strategist } = fixture;
+      const fromAmount = utils.parseEther("100");
+      const toAmount = utils.parseEther("100");
+
+      // Call swap method
+      const tx = oethVault
+        .connect(strategist)
+        .swapCollateral(weth.address, dai.address, fromAmount, toAmount, []);
+
+      await expect(tx).to.be.revertedWith("To asset is not supported");
+    });
+
+    it("Should swap if capital is paused", async () => {
+      const { weth, stETH, oethVault, strategist } = fixture;
+      const fromAmount = utils.parseEther("100");
+      const toAmount = utils.parseEther("100");
+
+      // Fund Vault with some assets
+      const vaultSigner = await impersonateAndFundContract(oethVault.address);
+      await weth.connect(vaultSigner).mint(fromAmount);
+
+      await oethVault.connect(strategist).pauseCapital();
+
+      // Call swap method
+      const tx = await oethVault
+        .connect(strategist)
+        .swapCollateral(weth.address, stETH.address, fromAmount, toAmount, []);
+
+      expect(tx).to.emit(oethVault, "Swapped");
+    });
+
+    it("Should revert if not called by Governor or Strategist", async () => {
+      const { weth, stETH, oethVault, josh } = fixture;
+      const fromAmount = utils.parseEther("100");
+      const toAmount = utils.parseEther("100");
+
+      // Call swap method
+      const tx = oethVault
+        .connect(josh)
+        .swapCollateral(weth.address, stETH.address, fromAmount, toAmount, []);
+
+      await expect(tx).to.be.revertedWith(
+        "Caller is not the Strategist or Governor"
+      );
+    });
+  });
+
+  describe("1inch Swapper", () => {
+    let fixture;
+
+    beforeEach(async () => {
+      fixture = await run1InchFixture();
+    });
+
+    it("Should swap assets using 1inch router", async () => {
+      const { swapper1Inch, strategist, weth, frxETH, mock1InchSwapRouter } =
+        fixture;
+
+      const deadAddr = "0x1111111111222222222233333333334444444444";
+
+      const data = utils.defaultAbiCoder.encode(
+        ["bytes4", "address", "bytes"],
+        [utils.arrayify("0x12aa3caf"), deadAddr, utils.arrayify("0xdead")]
+      );
+
+      const fromAmount = utils.parseEther("100");
+      const toAmount = utils.parseEther("100");
+
+      await weth
+        .connect(strategist)
+        .mintTo(swapper1Inch.address, fromAmount.mul(2));
+      await frxETH
+        .connect(strategist)
+        .mintTo(mock1InchSwapRouter.address, toAmount.mul(2));
+
+      await swapper1Inch.approveAssets([weth.address]);
```

Comment:
good question. It's because `Mock1InchSwapRouter` was only transferring assets in `swap` and not the Uniswap functions `unoswapTo` and `uniswapV3SwapTo`.
I'll add the transfers to the Uniswap functions to be consistent.

Reply:
aha I see thanks